src/Tokenizer/Analyzer/Analysis/ArgumentAnalysis.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tokenizer\Analyzer\Analysis;
+
+/**
+ * @internal
+ */
+final class ArgumentAnalysis
+{
+    /**
+     * The default value of the argument.
+     *
+     * @var null|string
+     */
+    private $default;
+
+    /**
+     * The name of the argument.
+     *
+     * @var string
+     */
+    private $name;
+
+    /**
+     * The index where the name is located in the supplied Tokens object.
+     *
+     * @var int
+     */
+    private $nameIndex;
+
+    /**
+     * The type of the argument.
+     *
+     * @var null|TypeAnalysis
+     */
+    private $type;
```

Comment:
this is not `type`, name is misleading

Reply:
`$typeAnalysis` feels a bit too bloated don't you think?

Reply:
no, as one, by just looking at name, would expect to have real argument here, not sth that will provide him info about indexes and is it scalar or not, without way to actually get that type.